### PR TITLE
[2361] Clean up and expand API V3

### PR DESCRIPTION
**Pull request by dankmitchell**
### Context
An API with no authentication - manage courses backend V3

### Changes proposed in this pull request
Add an index and show action for `providers` and `courses`.

### Guidance to review
- `http://localhost:3001/api/v3/recruitment_cycles/2020/providers`
- `http://localhost:3001/api/v3/recruitment_cycles/2020/providers/T92`
- `http://localhost:3001/api/v3/recruitment_cycles/2020/providers/T92/courses`
- `http://localhost:3001/api/v3/recruitment_cycles/2020/providers/T92/courses/X130`

### Checklist
- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by dankmitchell** [comment]
[API V3](https://trello.com/c/Mb1R2R6H/2456-api-v3)

**Comment by dankmitchell** [comment]
> > An open API aka manage courses backend V3
> 
> wouldn't you need something like this for the open api...
> https://github.com/slate-studio/openapi-rails
> 
> or am I just not reading this proper...

Re-worded the description.

File: config/routes.rb [hunk 1]
```diff
@@ -1,87 +1,123 @@
...
-#                                                                       PUT    /api/v2/access_requests/:id(.:format)                                                                                            api/v2/access_requests#update
-#                                                             error_500 GET    /error_500(.:format)                                                                                                             error#error_500
-#                                                            error_nodb GET    /error_nodb(.:format)                                                                                                            error#error_nodb
+# logstash not configured, falling back to standard Rails logging
+# Raven 2.12.2 configured not to capture errors: DSN not set
+# DEPRECATION WARNING: Using `bin/rake routes` is deprecated and will be removed in Rails 6.1. Use `bin/rails routes` instead.
+#  (called from load at /Users/Dan/.rbenv/versions/2.6.1/bin/rake:23)
```
Comment (defong) [inline config/routes.rb]: :roll_eyes:

Comment (dankmitchell) [reply config/routes.rb]: This was autogenerated so ive manually removed.

File: app/controllers/api/v3/courses_controller.rb [hunk 2]
```diff
@@ -3,9 +3,15 @@ module V3
...
+      def index
+        @courses = @provider.courses
+        render jsonapi: @courses, include: params[:include]
+      end
 
       def show
+        @course = @provider.courses.find_by!(course_code: params[:code].upcase)
+
         if @course.is_published?
           render jsonapi: @course, fields: fields_param
```
Comment (tvararu) [inline app/controllers/api/v3/courses_controller.rb]: Can you add a `params[:include]`? I think we / Apply need it to request locations

Comment (dankmitchell) [reply app/controllers/api/v3/courses_controller.rb]: Done.

**Pull request merged**